packages/zero-solid/src/use-query.test.tsx

```diff
@@ -194,6 +198,401 @@ test('useQuery query deps change', async () => {
   expect(resultDetailsLog).toEqual([{type: 'complete'}]);
 });
 
+test('useQuery query deps change, reconcile minimizes reactive updates', async () => {
+  const {tableQuery, queryDelegate} = setupTestEnvironment();
+
+  const [querySignal, setQuery] = createSignal(tableQuery.where('a', 1));
+
+  const renderHookResult = useQueryWithZeroProvider(
+    'useQuery-deps-change-id',
+    querySignal,
+  );
+
+  const [rows, resultDetails] = renderHookResult.result;
+
+  const row0Log: unknown[] = [];
+  const row1Log: unknown[] = [];
+  const resultDetailsLog: unknown[] = [];
+  const resetLogs = () => {
+    row0Log.length = 0;
+    row1Log.length = 0;
+    resultDetailsLog.length = 0;
+  };
+
+  runWithOwner(renderHookResult.owner, () => {
+    createEffect(() => {
+      row0Log.push(rows()[0]);
+    });
+    createEffect(() => {
+      row1Log.push(rows()[1]);
+    });
+    createEffect(() => {
+      const details = resultDetails();
+      resultDetailsLog.push({...details});
+    });
+  });
+
+  expect(row0Log).toEqual([
+    {a: 1, b: 'a', [refCountSymbol]: 1, [idSymbol]: '1'},
+  ]);
+  expect(row1Log).toEqual([undefined]);
+  expect(resultDetailsLog).toEqual([{type: 'unknown'}]);
+  resetLogs();
+
+  queryDelegate.gotCallbacks.forEach(cb => cb?.(true));
+  await 1;
+
+  expect(row0Log).toEqual([]);
+  expect(row1Log).toEqual([]);
+  expect(resultDetailsLog).toEqual([{type: 'complete'}]);
+  resetLogs();
+
+  // eslint-disable-next-line @typescript-eslint/no-floating-promises
```

Comment:
I don't know why the linter is getting this wrong... this method does not return a promise.